Content.Server/Chemistry/ReagentEffects/ChemVomit.cs

```diff
@@ -11,12 +13,12 @@ namespace Content.Server.Chemistry.ReagentEffects
     [UsedImplicitly]
     public sealed partial class ChemVomit : ReagentEffect
     {
-        /// How many units of thirst to add each time we vomit
-        [DataField]
-        public float ThirstAmount = -8f;
-        /// How many units of hunger to add each time we vomit
-        [DataField]
-        public float HungerAmount = -8f;
+        /// How many units of satiation to add each time we vomit
+        [DataField(customTypeSerializer: typeof(PrototypeIdDictionarySerializer<float, SatiationTypePrototype>))]
+        public Dictionary<string, float> SatiationAmount = new() {
```

Comment:
Use protoid, float and get rid of the custom type serializer

Reply:
Done